### DISCUSSION TITLE
Importer not working - networking/environment variable issue?

### DISCUSSION THREAD

**Discussion by Frittenpause** [firefly-iii/firefly-iii#8970] [Firefly III] [answered]
Hello everyone,

I decided to ask here after my normal resources and searches have not yet gotten me anywhere. I am currently in the process of setting up my Server at home and, after some research, decided to go for Firefly because it seemed to be the nicest system.

I am running a Proxmox and one VM has Ubuntu server on it with Portainer. I followed the instructions to get it up and running. (https://docs.firefly-iii.org/how-to/firefly-iii/installation/portainer/)

I can log into the firefly iii system and the importer is successfully installed and giving me the startup screen.
<img width="937" alt="Screenshot 2024-06-14 at 15 14 35" src="https://github.com/firefly-iii/firefly-iii/assets/4592181/9465c8f5-f9fd-4fda-99b7-c2ef26253dc6">

The IP of the VM is 192.168.10.109, the port for Firefly is 81, for the importer 82. The Container IPs are as seen in the screenshot. 
I did try to follow the steps and created a client ID with the  callback http://192.168.10.109:82/callback but to no avail. I also tried the Container IPs and even localhost (which is said to be not working on docker according to the documentation).

The called URL when i try to enter the client ID in the importer is 

http://localhost/oauth/authorize?client_id=14&redirect_uri=http%3A%2F%2F192.168.10.109%3A82%2Fcallback&response_type=code&scope=&state=[...]&code_challenge=[...]&code_challenge_method=S256

this fails wit "Unable to connect".

In the Importer Container Logs i can find this:

> 2024-06-14 15:31:25] production.DEBUG: Now at App\Http\Controllers\TokenController::submitClientId  
> 
> [2024-06-14 15:31:25] production.DEBUG: Submitted data:  {"client_id":"14"} 
> 
> [2024-06-14 15:31:25] production.DEBUG: [a] Base URL is "http://app:8080"  
> 
> [2024-06-14 15:31:25] production.DEBUG: [b] Vanity URL is now "http://app:8080"  
> 
> [2024-06-14 15:31:25] production.DEBUG: [c] Vanity URL is now "http://localhost"  
> 
> [2024-06-14 15:31:25] production.DEBUG: Now in App\Http\Controllers\TokenController::redirectForPermission(request, "http://app:8080", "http://localhost", 14)  
> 
> [2024-06-14 15:31:25] production.DEBUG: Query parameters are {"client_id":14,"redirect_uri":"http://192.168.10.109:82/callback","response_type":"code","scope":"","state":"[...]","code_challenge":"-[...]","code_challenge_method":"S256"}

There are no logs on the  firefly iii container at the same time.

As far as I understand the importer tries to direct me to localhost and from there to the external IP outside the container. The IP redirection does not change, no matter what I entered as the redirect URL when creating the client in the firefly instance.

I am think I might have missed something while creating the Stack and have now some Variables pointing at the wrong address, but I am not able to pinpoint it. 

The system is completely new setup and I can easily redo the whole process, I just would like to know what I did wrong before I do so.

Thanks a lot and greetings from Japan

Nils

**Comment by Frittenpause** [comment]
While writing this down I did some unintentional rubberducking and did read the documentation just once more. I tried to leave out the URL to enforce the system to ask me about it instead of assuming one, that resulted in the importer to not work at all. I tried the IP address for the URL instead of app:8080, that also didnt work. I tried to set URL and Vanity URL both to the IP and that worked.

Not sure *why*, but I take it. If someone would be so nice to explain why this is the case I could sleep peacefully again :)

Best regards,

Nils

**Comment by Frittenpause** [comment]
I did have to reinstall the system due to my raid being fried and there is no backup.

However, Install went smoothly in general. I have a New problem though which I cannot explain.

Everything is installed and works, I have a Nextcloud on :80 so Firefly is :81 and the Importer .82 - When I give the Importer the Client ID I go straight to a 404 page. 

After playing around with the environment variables again and again I looked at the called URL which is  in this case:

http://192.168.10.5:82/192.168.10.5:81/oauth/authorize?client_id=8&redirect_uri=http%3A%2F%2F192.168.10.5%3A82%2Fcallback&response_type=code&scope=&state=FBV[...]&code_challenge=P[...]o&code_challenge_method=S256

I then proceeded to remove the first part http://192.168.10.5:82/ and just went with the rest, that does indeed work. 

It is a valid workaround, however if someone happens to see what the problem might be there it would be nice to get it fixed.

Thanks in advance.

**Reply by JC5** [reply, marked as answer]
Make sure you add "http://" to the Firefly III address. Not just the IP address.

**Reply by Frittenpause** [reply]
Thank you! Not sure why I didn't try that earlier :/

The installation now is at least working perfectly, now I just need to get everything set up :)
